local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -17,28 +17,35 @@ limitations under the License.
 package deleter
 
 import (
+	"fmt"
 	"path/filepath"
+	"time"
 
 	"github.com/golang/glog"
-	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/types"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
 
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/pkg/api/v1"
 )
 
-type Deleter interface {
-	// Cleanup and delete all its owned PVs that have been released
-	DeletePVs()
-}
+const (
+	pollInterval = 1 * time.Second
+	// TODO: is this too fast?
```

Comment:
I think because deletion and creation occur serially right now, this is fine.  There is no risk of the informer updating the cache in the middle of the second deletion attempt , and the creation happening in the middle.

Maybe in that case, I should not even wait for it to be deleted from the cache and let it delete twice if it takes too long the first time.  The main controller already sleeps for a few seconds between loops.